File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0 
        r = 1
        max_diff = 0
        while r<len(prices):
            diff = prices[r]-prices[l]
            if diff>max_diff:
                max_diff = diff
            if diff<0:
                l=r
            r+=1
        return max_diff

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int max_profit = 0;
        int left=0;
        int right=0;

        while (right<prices.length){
            int diff = prices[right]-prices[left];
            if (diff>max_profit){
                max_profit=diff;
            }
            if (diff<0) {
                left=right;
            }
            right++;
        }
        return max_profit;

    }
}